add EDR & Network event standardization doc (#1906)
SEKOIA-IO/documentation: _shared_content/ECS parsing standariziation for EDR.md

          "working_directory": "/usr/bin"
        },
        "command_line": "git clone https://github.com/redcanaryco/atomic-red-team.git",
        "working_directory": "/usr/bin/",
        "start": "2024-07-02T13:50:51Z",
        "end": "2024-07-02T13:51:26Z"
      },
      "log": {
        "hostname": "linux-vm"
      },
      "file": {
        "hash": {
          "md5": "675853ca01ec441df7a015b91a7e1272",
          "sha256": "29aa689f38158d2e8941fa54e436f0260890af31cecad1e8799e5c2df7bc1ecc"
        }
      },
      "threat": {
        "tactic": {
          "name": "Malware"
        },
        "technique": {
          "name": "Malicious File"
        },
        "indicator": {
          "description": "Detection, process would have been killed if related prevention policy setting was enabled."
        }
      },
      "related": {
        "hash": [
          "29aa689f38158d2e8941fa54e436f0260890af31cecad1e8799e5c2df7bc1ecc",
          "675853ca01ec441df7a015b91a7e1272"
        ],
        "ip": [
          "172.17.0.1"
        ],
        "user": [
          "root"
        ]
      }
    
```
    

> Following these standards ensures that the EDR event data is consistent and can be easily interpreted and analyzed across different systems. This standardization helps in providing a unified view of security events and enhances the capability to respond to incidents efficiently.

Reviewer: ```suggestion
!!! note
     Following these standards ensures that the EDR event data is consistent and can be easily interpreted and analyzed across different systems. This standardization helps in providing a unified view of security events and enhances the capability to respond to incidents efficiently.
```

| `process.name`          | The process name that triggered the event/alert.    | `malware.exe`                                    |
| `process.pid`           | The process ID that triggered the event/alert.      | `2345`                                           |
| `process.commandline`   | The command that launched the process.          | `sudo deluser toto`                              |
| `file.hash`             | The hash of the file related to the event.      | `abcdef1234567890`                               |
| `agent.id`              | The unique ID of the agent that detected the event. | `12313-21341623`                             |
| `agent.name`            | The name of the agent that detected the event.  | `harfanglab`                                     |
| `host.hostname`         | The hostname that generated the event.           | `user admin`                                     |
| `host.os.full`          | The OS of the host that generated the event.     | `Windows 11 Enterprise`, `Ubuntu 22.04`          |
| `host.domain`           | The domain of the host that generated the event. | `Workgroup`                                      |

> ## Examples of event parsing

Reviewer: ```suggestion

## Examples of event parsing
```

# 🌟 Event Standardization for EDR Systems

> To ensure consistent recognition and semantic interpretation of **EDR (Endpoint Detection and Response) events**, we adhere to the following standardization guidelines. The supported EDR systems include **SentinelOne, HarfangLab, CrowdStrike, Cybereason, Sophos, Stormshield, Symantec, Tehtris, TrendMicro, and WithSecure**.

Reviewer: ```suggestion
To ensure consistent recognition and semantic interpretation of EDR (Endpoint Detection and Response) events, we follow the standardization guidelines bellow. We recommend users to adopt the same guidelines for their EDR custom format. The already supported EDR systems include SentinelOne, HarfangLab, CrowdStrike, Cybereason, Sophos, Stormshield, Symantec, Tehtris, TrendMicro, and WithSecure.
```

        "image_name": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
        "commandline": "powershell.exe -Command New-LocalUser -Name User9549 -Password (ConvertTo-SecureString -AsPlainText \"Password123\" -Force)",
        "parent_image": "C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.12_3.12.1264.0_x64__qbz5n2kfra8p0\\python3.12.exe",
        "fake_ppid": 0,
        "pe_timestamp": "1974-07-29T16:18:42.000Z",
        "create_time": "2024-07-12T08:13:13.221Z",
        "signed": true,
        "ioc_matches": [],
        "log_type": "process"
      },
      "agent": {
        "version": "3.10.2",
        "hostname": "vm-windows",
        "ostype": "windows",
        "osversion": "10.0.22631",
        "distroid": null,
        "domain": null,
        "agentid": "1553619e-8271-4941-8e10-9153e99cc5dc",
        "dnsdomainname": null,
        "osproducttype": "Windows 11 Enterprise",
        "domainname": "WORKGROUP",
        "additional_info": {}
      },
      "detection_origin": "agent",
      "rule_name": "Local User Created via PowerShell",
      "alert_type": "sigma",
      "log_type": "alert"
    }

>   ```
> For example, the event characteristics are derived from the log_type and alert_subtype(`process.log_type`) fields in the raw message:
> * The fields `event.kind` and `event.dataset` are set to alert because the log_type in the raw event is alert.
> * The field `event.category` is set to 'process' and `event.type` is set to 'start' because the `process.log_type` field of the raw event is identified as process.
> 
> **ECS Fields After Parsing**
>     
> ```json

Reviewer: ```suggestion
  ```
For example, the event characteristics are derived from the log_type and alert_subtype(`process.log_type`) fields in the raw message:
* The fields `event.kind` and `event.dataset` are set to alert because the log_type in the raw event is alert.
* The field `event.category` is set to 'process' and `event.type` is set to 'start' because the `process.log_type` field of the raw event is identified as process.

**ECS Fields After Parsing**
    
```json
```

# 🌟 Event Standardization for EDR Systems

To ensure consistent recognition and semantic interpretation of EDR (Endpoint Detection and Response) events, we follow the standardization guidelines bellow. We recommend users to adopt the same guidelines for their EDR custom format. The already supported EDR systems include SentinelOne, HarfangLab, CrowdStrike, Cybereason, Sophos, Stormshield, Symantec, Tehtris, TrendMicro, and WithSecure.

## 📋 Recommended Fields

> | **Field**        | **Description**                                | **Examples of values**                          |
> |------------------|------------------------------------------------|-------------------------------------------------|
> | `action.type`    | Specifies the action taken by the EDR system  | `block`, `allow`, `alert`                       |
> | `event.category` | The high-level category of the event          | `malware`, `intrusion`, `policy`                |
> | `event.code`     | A unique identifier or code representing the event | `12345`, `67890`, `ABCDE`                     |
> | `event.kind`     | Describes the kind of event                    | `event`, `alert`, `metric`                      |
> | `event.dataset`  | Specifies the dataset used by the EDR system  | `sentinelone`, `harfanglab`, `crowdstrike`, `cybereason`, `sophos`, `stormshield`, `symantec`, `tehtris`, `trendmicro`, `withsecure` |
> | `event.severity` | The severity level of the event               | `low`, `medium`, `high`, `critical`             |
> 
> ## 🛠️ Optional Fields
> 
> To provide more context and enhance the description of the events, including additional fields is encouraged. The following optional fields can be used:
> 
> | **Field**               | **Description**                                 | **Examples**                                     |
> |-------------------------|-------------------------------------------------|--------------------------------------------------|
> | `host.name`             | Name of the host where the event occurred       | `host-123`                                       |
> | `user.id`               | The unique identifier of the user involved in the event | `user-456`                                |
> | `process.name`          | The process name that triggered the event/alert.    | `malware.exe`                                    |
> | `process.pid`           | The process ID that triggered the event/alert.      | `2345`                                           |
> | `process.commandline`   | The command that launched the process.          | `sudo deluser toto`                              |
> | `file.hash`             | The hash of the file related to the event.      | `abcdef1234567890`                               |
> | `agent.id`              | The unique ID of the agent that detected the event | `12313-21341623`                             |
> | `agent.name`            | The name of the agent that detected the event   | `harfanglab`                                     |
> | `host.hostname`         | The hostname that generated the event           | `user admin`                                     |
> | `host.os.full`          | The OS of the host that generated the event     | `Windows 11 Enterprise`, `Ubuntu 22.04`          |
> | `host.domain`           | The domain of the host that generated the event | `Workgroup`                                      |

Reviewer: ```suggestion
| **Field**        | **Description**                                | **Examples of values**                          |
|------------------|------------------------------------------------|-------------------------------------------------|
| `action.type`    | Specifies the action taken by the EDR system.  | `block`, `allow`, `alert`                       |
| `event.category` | The high-level category of the event.          | `malware`, `intrusion`, `policy`                |
| `event.code`     | A unique identifier or code representing the event. | `12345`, `67890`, `ABCDE`                     |
| `event.kind`     | Describes the kind of event.                    | `event`, `alert`, `metric`                      |
| `event.dataset`  | Specifies the dataset used by the EDR system.  | `sentinelone`, `harfanglab`, `crowdstrike`, `cybereason`, `sophos`, `stormshield`, `symantec`, `tehtris`, `trendmicro`, `withsecure` |
| `event.severity` | The severity level of the event.               | `low`, `medium`, `high`, `critical`             |

## 🛠️ Optional Fields

To provide more context and enhance the description of the events, including additional fields is encouraged. The following optional fields can be used:

| **Field**               | **Description**                                 | **Examples**                                     |
|-------------------------|-------------------------------------------------|--------------------------------------------------|
| `host.name`             | Name of the host where the event occurred.       | `host-123`                                       |
| `user.id`               | The unique identifier of the user involved in the event. | `user-456`                                |
| `process.name`          | The process name that triggered the event/alert.    | `malware.exe`                                    |
| `process.pid`           | The process ID that triggered the event/alert.      | `2345`                                           |
| `process.commandline`   | The command that launched the process.          | `sudo deluser toto`                              |
| `file.hash`             | The hash of the file related to the event.      | `abcdef1234567890`                               |
| `agent.id`              | The unique ID of the agent that detected the event. | `12313-21341623`                             |
| `agent.name`            | The name of the agent that detected the event.  | `harfanglab`                                     |
| `host.hostname`         | The hostname that generated the event.           | `user admin`                                     |
| `host.os.full`          | The OS of the host that generated the event.     | `Windows 11 Enterprise`, `Ubuntu 22.04`          |
| `host.domain`           | The domain of the host that generated the event. | `Workgroup`                                      |
```

## Examples of event parsing

> In this section, raw events extracted from real use cases are used to show the expected parsing outcome . For each example, the input will be a raw event in json format and the output will be in ECS.

Reviewer: ```suggestion
In this section, raw events extracted from real use cases are used to show the expected parsing outcome. For each example, the input will be a raw event in json format and the output will be in ECS.
```

        "end": "2024-07-02T13:51:26Z"
      },
      "log": {
        "hostname": "linux-vm"
      },
      "file": {
        "hash": {
          "md5": "675853ca01ec441df7a015b91a7e1272",
          "sha256": "29aa689f38158d2e8941fa54e436f0260890af31cecad1e8799e5c2df7bc1ecc"
        }
      },
      "threat": {
        "tactic": {
          "name": "Malware"
        },
        "technique": {
          "name": "Malicious File"
        },
        "indicator": {
          "description": "Detection, process would have been killed if related prevention policy setting was enabled."
        }
      },
      "related": {
        "hash": [
          "29aa689f38158d2e8941fa54e436f0260890af31cecad1e8799e5c2df7bc1ecc",
          "675853ca01ec441df7a015b91a7e1272"
        ],
        "ip": [
          "172.17.0.1"
        ],
        "user": [
          "root"
        ]
      }
    
```
    
Following these standards ensures that the EDR event data is consistent and can be easily interpreted and analyzed across different systems. This standardization helps in providing a unified view of security events and enhances the capability to respond to incidents efficiently.

>     
> As for the special fields that might exist from an EDR to another. The adopted parsing is by giving specific ECS field with different subfields that caracterize the EDR. 
>     

Reviewer: ```suggestion
```
